extensions-contrib/accurate-cardinality/src/main/java/org/apache/druid/query/aggregation/cardinality/accurate/bitmap64/ImmutableBitmap.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.query.aggregation.cardinality.accurate.bitmap64;
+
+import org.roaringbitmap.longlong.LongIterator;
+
+/**
+ * This class is meant to represent a simple wrapper around an immutable bitmap
+ * class.
+ */
+public interface ImmutableBitmap
```

Comment:
Likewise, suggest using `LongImmutableBitmap` or something like here to distinguish from processing library types.

Reply:
Actually, I've pulled a copy of this branch and.had a closer look and some time to think about it a bit. The only reason for all of this abstraction is if this extension needed to support using multiple bitmap implementations for computing the distinct count, but it only seems to be using roaring. I think it might be better to drop all of this extra structure and just using roaring bitmap directly, either into a concrete collector, also without all of the interface infrastructure, or just pull it all the way up into the aggregators themselves. I think this would significantly simplify the implementation as well.

Is there any reason to support multiple bitmap implementations that I am missing?